labs/b6.md

```diff
@@ -0,0 +1,144 @@
+
+First press `<F2>`, scroll down to Display options, and check "Hide userland process threads." We won’t be dealing with those in this lab.
+
+Now open up another terminal and SSH into your VM. Run the command `yes`. It uses a lot of resources as it prints a continuous stream of `y`'s. What resource specifically does the `yes` command exhaust? If you are having trouble finding this, press `<` to choose which resource to order processes by. Make sure to quit out of `yes` once you are finished.
+
+### The process hierarchy 
+Run `htop` once more. This time click `<F5>` to enter Tree View. You should see a visual representation of the process hierarchy on your system, with everything stemming from `/sbin/init` (which is really just a symbolic link to **systemd**).  
+
+For curious students that are interested in seeing a more extensive process hierarchy on a large system, you are encouraged to run `htop` on the OCF server `tsunami`. Let us know of any cool processes that you find! 
+
+### Orphan processes
+Open a second terminal and SSH to your VM. Now run `sleep 1000 &`. You should see this new process pop into your `htop` session on your first terminal. If not, press `<F3>` and search for "sleep." What is its parent?
+
+Select this parent and press `<F9>` to kill it. Send the `SIGTERM` signal. The sleep process now has `init` as its new parent, which is PID 1. What you just did is manually orphan a process; when that happens said process is subsequently re-parented by the `init` process. 
+
+Now go through the same process again. This time, send the parent a `SIGHUP` (hangup) signal. Can you still find the sleep process? When `SIGHUP` is sent to a parent shell, the parent subsequently sends hangup signals to any child processes before terminating; all processes that receive `SIGHUP` from a parent shell will terminate -- this is one way to avoid creating orphan processes.
+
+If you are interested in learning about the different signals, run `man 7 signal`. 
+
+## Cron
+So much infrastructure in the computing world relies on scheduled processes. This is the job of the cron daemon. For example, at the OCF, we use a cron job that runs every thirty minutes to keep our servers and desktops updated via puppet. To get a feel for the cron scheduler, we're going to write a basic cron job.
+
+Open the cron editor by running `crontab -e` (if the editor of your choice isn't being launched, set the `EDITOR` environment variable), which will create a crontab for your user. Here is a sample task:
+
+```* * * * * date +"\%T" >> $HOME/timestamps.txt```
+
+Right now this runs every minute. Modify it to run every five minutes. Make sure to leave this running while you complete the remainder of the lab as you will be turning this file in.
+
+If you get stuck, visit crontab.guru!
+
+## Using systemd
+
+### What services are running right now? 
+Run `systemctl`. You’ll see a long table of every unit known to systemd.
```

Comment:
We could say "everything" instead of "every unit", unless we're pretty confident that every student will remember what a systemd unit is (otherwise they might get confused by the wording). Sorry, I'm starting to get kinda nitpicky.